vsintegration/src/FSharp.Editor/CodeFix/IFSharpCodeFix.fs

```diff
@@ -7,5 +7,12 @@ open Microsoft.CodeAnalysis.Text
 
 open CancellableTasks
 
-type IFSharpCodeFix =
-    abstract member GetChangesAsync: document: Document -> span: TextSpan -> CancellableTask<string * TextChange list>
+type FSharpCodeFix =
+    {
+        Name: string
+        Message: string
+        Changes: TextChange list
+    }
+
+type IFSharpCodeFixProvider =
+    abstract member GetCodeFixIfAppliesAsync: document: Document -> span: TextSpan -> CancellableTask<FSharpCodeFix option>
```

Comment:
I vaguely remember that you will need a list of codefixes for a single invocation eventually => that is, after you have a look at the more complex code fixes as well.

Have a look e.g. at removed unused binding - a single codefix offers both replacing as well as prefixing with an underscore (= two possible codefix actions coming out of it).

Reply:
Yes, good point, we might end up there. I gradually address more complex code fixes - eventually we'll either generalize this signature further or maybe somehow split/refactor harder cases.